---
start: true
title: "Introduction"
exercises: 30
teaching: 20
questions:
- "Where does the dataset come from?"
- "How to login"
- "Where are the files located"
objectives:
- "Understand the data"
- "Choose login details"
- "Familiarize yourself with the environment"
keypoints:
- "Sequencing *S. pneumoniae* patient isolates to determine assocations of bacterial genes with disease severity"
---

## Introduction

We will be making use of Google Docs: [https://docs.google.com/document/d/19RkKurPgTVAOrzZINQBKUjb2CkMRrsxjejERBvUJCq0/edit](https://docs.google.com/document/d/19RkKurPgTVAOrzZINQBKUjb2CkMRrsxjejERBvUJCq0/edit).
The Google doc contains the link to the shared documents, the IP address of the server, etc. 

### Dataset

Introduction to the dataset is given here: [https://www.dropbox.com/s/hi7ivyml0vle2k1/Bacterial%20GWAS%20DEC%202018.pptx?dl=0](https://www.dropbox.com/s/hi7ivyml0vle2k1/Bacterial%20GWAS%20DEC%202018.pptx?dl=0) . In total we will be analyzing 62 genomes, of which one is a closed reference genome (OXC141). 

### How to login

Follow the instructions at <google docs file>.
The server we will be using is named klif2.uu.nl, ipaddress: 131.211.18.6. We will be using a shared server as the tools used for this part of the course require more computing power and small single Amazon instances will not be sufficient. As a consequence, everyone has their own username/password (see the Google Docs file). It is possible to access the server using our Jupyterhub system [https://klif2.uu.nl](https://klif2.uu.nl), however use of MobaXterm is recommended. 

### Where are the files located

In your home folder (~/), you will find the directory "reads" which contains symlinks to the read files used in this study. As assembling of all the genomes in this study would be too time consuming, we will assembling only two genomes per person. We will combined the outputs of each person later on.
  
  
